gnovm/pkg/gnolang/preprocess.go

```diff
@@ -154,12 +154,6 @@ func Preprocess(store Store, ctx BlockNode, n Node) Node {
 
 		defer func() {
 			if r := recover(); r != nil {
-				fmt.Println("--- preprocess stack ---")
```

Comment:
Are you removing a duplicate or something that was only accessible here? Can you show a before/after?

Reply:
before:
```
10:47:05 ~/lint_example $ gno lint --root-dir=~/code/gnolang/gno .
./.: missing 'gno.mod' file (code=1).
--- preprocess stack ---
stack 2: func Render(path (const-type string)) { return (const ("hello" <untyped> string)) }
stack 1: file{ package lint; import fmt fmt; func Render(path (const-type string)) { return (const ("hello" <untyped> string)) } }
stack 0: package(lint)
------------------------
lint.gno:6: expected 0 return values; got 1 (code=2).
```

after:
```
10:48:19 ~/lint_example $ gno lint --root-dir=~/code/gnolang/gno .
./.: missing 'gno.mod' file (code=1).
lint.gno:6: expected 0 return values; got 1 (code=2).
```

For the linter to be properly usable, I need to either:
- Comment or remove those lines.
- Redirect stdout/stderr to `/dev/null` to prevent them from being altered.

Perhaps a good middle ground would be to comment them ?
```diff
modified   gnovm/pkg/gnolang/preprocess.go
@@ -154,6 +154,14 @@ func Preprocess(store Store, ctx BlockNode, n Node) Node {
 
 		defer func() {
 			if r := recover(); r != nil {
+				// Uncomment this to debug preprocess stack.
+				// fmt.Println("--- preprocess stack ---")
+				// for i := len(stack) - 1; i >= 0; i-- {
+				// 	sbn := stack[i]
+				// 	fmt.Printf("stack %d: %s\n", i, sbn.String())
+				// }
+				// fmt.Println("------------------------")
+
 				// before re-throwing the error, append location information to message.
 				loc := last.GetLocation()
 				if nline := n.GetLine(); nline > 0 {
```

Reply:
I think the "preprocess stack" log needs to be kept. (And I am of the opinion that since Git exists, commenting code is NEVER a good idea :P)

Two ideas (I prefer the first):

- adding the preprocess stack to the error message (from the panic below), possibly in a custom type so lint can catch it and not print it
- using `if debug` / `debug.Println`

Reply:
Good idea ! I've added a `PreprocessStackError` custom error to encapsulate error and preprocess stack